### ISSUE TITLE
Support command line window editing

### ISSUE THREAD

**Issue by haifengkao** [VSCodeVim/Vim#4007]
**Is your feature request related to a problem? Please describe.**
Support command line window
command line window allows you to do normal mode editing on command line histories.
https://vim.fandom.com/wiki/Using_command-line_history

**Describe the solution you'd like**
![螢幕快照 2019-08-28 下午4 27 23](https://user-images.githubusercontent.com/4080524/63838856-c4852d00-c9b0-11e9-9279-5e66db1cdc07.jpg)

**Describe alternatives you've considered**
The current VSCodeVim can active the command line window by `q:`
I can select the old commands, but I cannot edit them.
![螢幕快照 2019-08-28 下午4 28 24](https://user-images.githubusercontent.com/4080524/63838919-e1216500-c9b0-11e9-9561-003dbb81cc61.jpg)

**Comment by stevenguh** [comment]
I don't think the QuickPick API from VS Code alone can provide that functionality, but we can combine with InputBox to provide editing of the command.

When the user pick from the QuickPick API, we will need to prompt another InputBox for the user to edit the command if needed. However, this will add one extra enter press before executing that command if the user didn't want any editing. It's similar to the example project below
https://github.com/Microsoft/vscode-extension-samples/tree/master/quickinput-sample

**Comment by J-Fields** [comment]
Yeah, we could use QuickPick+InputBox to allow you to edit the one you've picked, but to get the full modal editing experience vim gives you with this feature, we'd have to do some janky stuff
